WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -234,6 +235,12 @@ private ProcessedResult ProcessNoLock(SmartTransaction tx)
 				// If we did not have it.
 				if (Coins.TryAdd(newCoin))
 				{
+					// If the coin was banned from coinjoining before.
+					if (Coins.IsCoinBanned(newCoin, DateTimeOffset.Now, out DateTimeOffset? bannedUntil))
+					{
+						newCoin.BannedUntilUtc = bannedUntil;
+					}
```

Comment:
Isn't this whole PR's idea to get rid of the SmartCoin's `BannedUntil` and `IsBanned` property?
Then why do we assign stuff to the SmartCoin's properties?

Reply:
1. The TransactionProcessor processes TXs upon loading the wallet, so it can ban coins which have been banned before upon loading.
But I agree that this is **not** the good place to deal with it, **I'll change it**.
2. Yes, we wan't to get rid of `IsBanned` for sure, but `BannedUntil` is a little bit tricky, as we need to display the date time on the Ctrl+C+D coins view panel.